Publish TWIM (#2684)
matrix-org/matrix.org: content/blog/2025/04/2025-04-11-twim.md

+++
date = "2025-04-11T15:30:00Z"
title = "This Week in Matrix 2025-04-11"
path = "/blog/2025/04/11/this-week-in-matrix-2025-04-11"

[taxonomies]
author = ["Thib"]
category = ["This Week in Matrix"]
+++

## Matrix Live

> 

Reviewer: ```suggestion

Migrating millions of matrixers to MAS

```

Reviewer: hm or this goes into the title like `## Matrix Live S10E21 — Migrating millions of matrixers to MAS`

> 
> On Monday this week, the backend team at Element successfully migrated Matrix.org to use [matrix-authentication-service (MAS)](https://github.com/element-hq/matrix-authentication-service).  
> This happened right after the [core MSCs for next-generation auth](https://github.com/matrix-org/matrix-spec-proposals/pull/3861) got merged into the Matrix spec!
> 
> After a full week of running MAS on Matrix.org, we're happy to call the migration a great success! 🕺
> 
> This means Matrix.org users now have a new home to manage their accounts: [account.matrix.org](https://account.matrix.org/)
> 
> You can read more about the migration on the Matrix.org blog:  
> 👉 [Matrix.org is now running MAS!](https://matrix.org/blog/2025/04/morg-now-running-mas/)  
> ...and what it means for Element users on the Element blog:  
> 👉 [MAS migration unleashes Element X on Matrix.org](https://element.io/blog/mas-migration-unleashes-element-x-on-matrix-org/)
> 
> We're still ironing out a few bugs, so if you’ve experienced any issues related to the migration, let us know in [#matrix-auth:matrix.org](https://matrix.to/#/#matrix-auth:matrix.org)!

<!-- more -->

## Dept of Academia 🎓️

> 
> [carla](https://matrix.to/#/@lalider:matrix.org) says
> 
> > __PhD position in Human-Computer Interaction Principles for Messaging Interoperability available in Copenhagen, Denmark!__

Reviewer 2: ```suggestion
### PhD position in Human-Computer Interaction Principles for Messaging Interoperability available in Copenhagen, Denmark!

[carla](https://matrix.to/#/@lalider:matrix.org) says

```

>
> A unique opportunity for a fully funded 3-year position to design user-friendly interfaces for managing privacy in interoperable chats.
> 
> The PhD position is in the area of Human-Computer Interaction, supervised by me (Carla Griggio) and Prof. Niels van Berkel at the Department of Computer Science in Aalborg University (Copenhagen Campus!).
> 
> Development experience related to Matrix is a plus but not a requirement. The ideal candidate has solid software engineering skills and cares deeply about interoperability and/or the user experience of managing privacy in text messaging.
> 
> 🎓 You can find more info about the position and [apply here](<https://www.stillinger.aau.dk/phd-stillinger/vis-stilling/vacancyId/1219575>).
> 
> 🎦 This project is directly related to "the walled garden dilemma" I presented at the MatrixConf, and you can [watch the video here](<https://youtu.be/0OWAPYS2QIE?si=CUIo69rAN7LS3cze>).

> 
> [Florian Jacob](https://matrix.to/#/@bh7953:kit.edu) announces

Reviewer 2: ```suggestion
### Research Paper "ReP2P Matrix"

[Florian Jacob](https://matrix.to/#/@bh7953:kit.edu) announces
```

> Our research paper [“ReP2P Matrix: Decentralized Relays to Improve Reliability and Performance of Peer-to-Peer Matrix”](https://dl.acm.org/doi/abs/10.1145/3694809.3700741), is now available under CC-BY, [code and data based on the existing peer-to-peer Matrix prototypes](https://gitlab.kit.edu/kit/tm/telematics/rep2p-matrix) is available as well. We thought of a concept for “disposable decentralized relays” - like simple, user-independent homeservers with minimal privileges and without users being bound to only one - and evaluated it as an additional way of message dissemination that complements the current P2P Matrix prototypes. The work was a cooperation between the [Telematics](https://telematics.tm.kit.edu/) and [Decentralized Systems](https://www.dsn.kastel.kit.edu/) research groups, and was presented at the [Decentralization of the Internet Workshop](https://conferences.sigcomm.org/co-next/2024/#!/din) at [ACM CoNext](https://conferences.sigcomm.org/co-next/2024/#!/home).

## Dept of Clients 📱

> 
> 

Reviewer: ```suggestion
```

A total rewrite of Element-iOS using the Matrix Rust SDK underneath and targeting devices running iOS 16+.

[Doug](https://matrix.to/#/@douge:matrix.org) reports

> This week we released Element X 25.04.3 with a collection of improvements such as
> 
> * The option to only hide timeline media in public rooms.
> * The ability to kick/ban someone directly from the timeline (and add a reason when you do it).
> * Improvements to hopefully reduce the amount of space use by both the SDK and the app.
> 
> On top of this we’ve continued to work on support for Threads, added the ability to enter your PIN with a keyboard and improved the smoothness of the authentication flow when using OIDC.

### Element X Android ([website](https://github.com/vector-im/element-x-android))

Android Matrix messenger application using the Matrix Rust SDK and Jetpack Compose.

[Benoit](https://matrix.to/#/@benoit.marty:matrix.org) says

> We are releasing a new RC 25.04.1 to fix a crash introduced recently that did not get its way to production. Thanks for all people who have sent rageshakes and to YamatoRyou for having created the issue!
> 

> > Besides that, we're working on improving accessibility, when using a screen reader. Also there have been some changes on the settings to prevent image from invitation to be visible for safety reason.

Reviewer: ```suggestion
> Besides that, we're working on improving accessibility when using a screen reader. Also there have been some changes on the settings to prevent images from invitations to be visible for safety reason.
```

> 
> You'll also notice that the app got a new icon!

### Cinny ([website](https://cinny.in))

Cinny is a Matrix client focused on simplicity, elegance and security

[Lozenge](https://matrix.to/#/@kfiven:matrix.org) announces

> Cinny v4.5.0 & v4.6.0: Enhanced room management and Privacy
> 

> > @twim:cadair.com The Cinny team is pleased to announce our latest updates, delivering powerful new tools for room management and user privacy.

Reviewer 2: ```suggestion
> The Cinny team is pleased to announce our latest updates, delivering powerful new tools for room management and user privacy.
```

> 
> Our v4.5.0 release focused on privacy controls, giving users the ability to disable read receipts and typing indicators, along with automatic hiding of messages from ignored users. We’ve also implemented image upload previews and streamlined deleted message handling.
> 
> These features are now easily accessible through our redesigned room/space settings interface.
> 
> Experience these updates today:
> 🌐 [app.cinny.in](https://app.cinny.in)
> 💬 Join our community: [#cinny-space:matrix.org](https://matrix.to/#/#cinny-space:matrix.org)
> 🗞️ Follow us on [Bluesky](https://bsky.app/profile/cinnyapp.bsky.social)
> 📝 Release notes: [GitHub](https://github.com/cinnyapp/cinny/releases)

### Tammy ([website](https://tammy.connect2x.de))

Multiplatform messengers build on top of Trixnity Messenger

Benedict announces

> Over the past few weeks, **Tammy** has grown in leaps and bounds with **tons of improvements** aimed at making Matrix more accessible, reliable, and delightful across platforms — especially on the web! Here’s a quick roundup of the journey from 1.1.3 to 1.1.11 (or, in dev-speak, from Trixnity Messenger `3.2.0` to `3.5.3` under the hood). 
> 

> > #### ✨ Major New Features
> > 
> > * **Custom Themes & Material 3 Styling**
> > 
> > Say hello to **dynamic theming**! Tammy now uses a brand-new Material 3 styling system, paving the way for creative themes — yes, even nostalgic ones like _Windows 9x_. This is just an idea, but maybe you’ll be able to apply themes just by joining a Matrix room in the future.
> > * **Archived Rooms Support**
> > 
> > Left a room but still want to check what happened? No problem — rooms stick around until you manually delete them.
> > * **Knocking Support**
> > 
> > We now support knocking into public rooms and accepting knocks in the member list.
> > * **Accessibility Upgrades**
> > 
> > You can now adjust **font and display sizes** to suit your visual preferences.
> > 
> > #### 🔧 Improvements & UI Enhancements
> > 
> > * **Revamped Styling**
> > 
> > Tammy’s UI has been polished with **unified styling** for buttons, surfaces, tooltips, floating action buttons, and input areas. No more flickers or mismatched elements.
> > * **User Profile Screen**
> > 
> > You can now view member profiles and do some room actions on them.
> > * **Performance Boosts**
> > 
> > Web experience is much smoother, because we fixed many bugs.
> > 
> > #### 🐛 Fixes You’ll Appreciate
> > 
> > * **Decryption & Verification Fixes**
> > 
> > Several bugs around device verification and message decryption are gone. **Note: You may need to re-login to ensure all keys sync correctly!**
> > * **Web: One Instance at a Time**
> > 
> > We added a check to ensure only one running instance of Tammy in your browser to avoid database corruption.

Reviewer 2: ```suggestion
> #### ✨ Major New Features
> 
> * **Custom Themes & Material 3 Styling**
> 
>   Say hello to **dynamic theming**! Tammy now uses a brand-new Material 3 styling system, paving the way for creative themes — yes, even nostalgic ones like _Windows 9x_. This is just an idea, but maybe you’ll be able to apply themes just by joining a Matrix room in the future.
> * **Archived Rooms Support**
> 
>   Left a room but still want to check what happened? No problem — rooms stick around until you manually delete them.
> * **Knocking Support**
> 
>   We now support knocking into public rooms and accepting knocks in the member list.
> * **Accessibility Upgrades**
> 
>   You can now adjust **font and display sizes** to suit your visual preferences.
> 
> #### 🔧 Improvements & UI Enhancements
> 
> * **Revamped Styling**
> 
>   Tammy’s UI has been polished with **unified styling** for buttons, surfaces, tooltips, floating action buttons, and input areas. No more flickers or mismatched elements.
> * **User Profile Screen**
> 
>   You can now view member profiles and do some room actions on them.
> * **Performance Boosts**
> 
>   Web experience is much smoother, because we fixed many bugs.
> 
> #### 🐛 Fixes You’ll Appreciate
> 
> * **Decryption & Verification Fixes**
> 
>   Several bugs around device verification and message decryption are gone. **Note: You may need to re-login to ensure all keys sync correctly!**
> * **Web: One Instance at a Time**
> 
>   We added a check to ensure only one running instance of Tammy in your browser to avoid database corruption.
```

## Dept of SDKs and Frameworks 🧰

### Trixnity ([website](https://gitlab.com/trixnity/trixnity))

Multiplatform Kotlin SDK for developing Clients, Bots, Appservices and Servers

Benedict reports

> Trixnity got some polish over the past few weeks. The sync loop has been completely reworked to behave more predictably, especially when switching filters or timeouts - great news for notification reliability. On top of that, client-side support for the Matrix 1.11 federation media API has landed. As usual, a bunch of bugs were squashed and minor API improvements made to keep things smooth and stable.

### matrix-rust-sdk ([website](https://github.com/matrix-org/matrix-rust-sdk))

Next-gen crypto-included SDK for developing Clients, Bots and Appservices; written in Rust with bindings for Node, Swift and WASM

[Ivan 🦀](https://matrix.to/#/@mnt_io:matrix.org) reports
 
> Time flies! Did you miss us? Because we missed you. Our latest update report was 3 weeks ago. We were pretty busy!
> 
> First off, we are happy to announce the immediate availability of [the Matrix Rust SDK 0.11 release][release] \o/. Feel free to check the `CHANGELOG.md` files of all the crates you're interested by.
> 

> > ### Thread & Timeline

Reviewer: ?
```suggestion
> ### Threads & Timeline
```

> 
> Right now, works as started [to send messages in threads][#4819], or [to refactor the `Timeline` to aggregate items together][#4839].
> 
> A notable change in the `Timeline`, unrelated to thread, has been introduced: [a new virtual timeline item to represent the start of the `Timeline`][#4816]. This is pretty useful if you want to render a special header at the start of the `Timeline`. The consumer of the `Timeline` no longer needs to do a computation dance to detect its start.
> 
> ### Media gallery
> 
> A new MSC has been added: [MSC4274, Inline media galleries via `msgtypes`][MSC4274]:
> 
> > Matrix allows sharing individual media files such as images or videos via
> > dedicated `m.room.message` `msgtypes`. This is, however, not practical for
> > sharing larger collections of files. For one thing, there is no mechanism
> > to set a joint caption for several media files. For another, there is no way
> > to inform other clients that a set of pictures belongs together and should be
> > grouped accordingly in the timeline. These shortcomings call for a built-in
> > media gallery feature as it is widely known from other messaging apps.
> 

> > As with all MSC, an implementation on a client and a homeserver is required. And here comes [a set of patches to implement this MSC inside the Matrix SDK][#4898].

Reviewer: ```suggestion
> As with all MSCs, an implementation on a client and a homeserver is required. And here comes [a set of patches to implement this MSC inside the Matrix SDK][#4898].
```

> 3. interactive verification workflow for new joiners: a new hip way to combat spam in your community rooms. you can now configure some or all of your rooms to require that new members verify they know how to accept a DM invite, and are capable of following simple instructions! migration command `!community verify-migrate` will fix your existing room's permissions safely (hopefully) so anyone already in your room doesn't need to verify.
> 4. bulk message redaction: redact up to the last 100 messages from a user in a room on demand, or enable automatic bulk redaction in all your community rooms when a user is banned with the `redact_on_ban` setting. _finally_.
> 
> feel free to join my development channel and see first-hand how the interactive verification workflow works, ask questions (after you verify 😉), get help, or suggest features! [#dev:mssj.me](https://matrix.to/#/#dev:mssj.me)

## Dept of Trust & Safety 🙆

> [Jim](https://matrix.to/#/@mackesque:matrix.org) announces

Reviewer: Hi @jimmackenzie, thanks for your update! I appreciate the guidance about the hide images setting a lot since it's not always easy to find. When speaking neutrally on behalf of the Foundation, I would love if we could get closer to the ideal of including the info for clients beyond the element clients, perhaps starting with these we recommend on the website https://matrix.org/ecosystem/clients/. I appreciate if the T&S team has no capacity to research this at this moment. It would probably good to have an overview of T&S features in popular clients anyway both for the team and the public (https://github.com/matrix-org/matrix.org/issues/2685). If you or anyone would happen to have such info at hand and are able to share it (even in parts), we would be happy to add it on to this TWIM.

Reviewer 3: I would **very much** like to include more inclusive guidance for all the clients in the ecosystem. That would be an excellent project for the T&S Working Group of the Governing Board (and for the folks who are building clients, I'd love to hear about your work on safety tooling).

> ### Trust & Safety update
> 
> Hi folks. You may have noticed a rash of attacks on public Matrix rooms, where attackers send floods of illegal imagery, designed to shock and appall.  We haven’t been commenting publicly as it’s an ongoing incident, but we want to give you a brief update:
> 

> > * The attackers are overloading homeservers with malicious federation traffic to slow down our ability to redact messages (giving the impression that we’re not responding to the raids: we are).  We’re working on solutions to this.  

Reviewer: ```suggestion
> * The attackers are overloading homeservers with malicious federation traffic to slow down our ability to redact messages (giving the impression that we’re not responding to the raids: we are). We’re working on solutions to this.  
```

> * We’re also working on ways for servers to filter abusive content before it ever gets sent to clients; we’ll publish details as soon as we can..  

> > * Whilst this is ongoing, we strongly recommend hiding image previews  in your client.  

Reviewer: ```suggestion
> * Whilst this is ongoing, we strongly recommend hiding image previews in your client.  
```
